src/components/dev/LocalizationReport.js

```diff
@@ -0,0 +1,226 @@
+import React, { Component } from 'react';
+import {  
+  Button,
+  Dropdown,
+  Form,
+  Icon,
+  List,
+  Menu,
+  Modal,
+} from 'semantic-ui-react';
+import _ from 'lodash';
+import { getKeyPathsArray } from '../../utils/objectHelper';
+
+// DATA
+import { localizations } from '../../localization/all';
+
+
+const ResultsList = (props) => {
+  
+  const { results, filter } =  props;
+
+  const filterResults = (results, filter) => {
+    if (filter === 'all') { 
+      return results;
+    }
+    let pass = filter === 'true' ? true : false;
+    return results.filter((result) => {
+      return result.pass === pass;
+    });
+  };
+
+  let filteredResults = filterResults(results, filter);
+
+  return filteredResults.map((filteredResult) => {
+    return (
+      <List.Item
+        key={ filteredResult.keyPath }>
+        <List.Icon
+          name={ filteredResult.pass ? 'check square' : 'window close' }
+          size='large'
+          color={ filteredResult.pass ? 'green' : 'red' }
+          verticalAlign='middle' />
+        <List.Content>
+          { filteredResult.keyPath } { filteredResult.test } '{ filteredResult.in }'.
+        </List.Content>
+      </List.Item>
+    );
+  });
+};
+
+
+class LocalizationReport extends Component {
+
+  state = {
+    modalOpen:        false,
+    modelLocKey:      null,
+    compareLocKey:    null,
+    filter:           'all',
+    localizationKeys: [],
+  };
+
+  componentDidMount() {
+    // Get the list of non-EN localizations for testing
+    let localizationKeys = Object.keys(localizations);
+    const enIndex = localizationKeys.indexOf('en');
+
+    // Initilize out state if EN localization exists
+    if (enIndex !== -1) {
+      // Remove EN from our keys to loop over
+      localizationKeys.splice(enIndex, 1);
+
+      this.setState({
+        modelLocKey:   'en',
+        compareLocKey: localizationKeys[ 0 ],
+        localizationKeys,
+      });
+    }
+  }
+
+  toggleModalOpen = () => {
+    let modalOpen = this.state.modalOpen;
+    this.setState({ modalOpen: !modalOpen });
+  };
+
+  setLanguage = (e, { value }) => { this.setState({ compareLocKey: value }); };
+  
+  setFilter = (e, { value }) => { this.setState({ filter: value }); };
+ 
+  compareLocalizations = (modelLocKey, compareLocKey) => {
+    let results = [];
+
+    if (modelLocKey !== null && compareLocKey !== null) {
+      // Get the key paths from the model (EN) localization
+      const modelKeyPaths = getKeyPathsArray(localizations[ modelLocKey ]);
+     
+      // Loop through all non-en localizations
```

Comment:
This comment is outdated; maybe "Loop through all paths" instead?

Reply:
Good catch.  That is indeed inaccurate at this point.  This should do?  // Loop through all model (EN) key paths